### PR TITLE
Back paginate list of recipes

### PR DESCRIPTION
**Pull request by SamiDghim**
add gem will_paginate to paginate recipes

update seed.rb file to fill the data base

activate the pagination of list recipes

### CODE REVIEW

**Comment by SamiDghim** [comment]
@melas123 @achr3f @insatmc @thameursaadi

File: app/controllers/recipe_controller.rb [hunk 1]
```diff
@@ -7,8 +7,9 @@ class RecipeController < ApplicationController
   respond_to :json
   #Show all recipes
   def index
-    @recipes = Recipe.all
-    respond_with( @recipes )
+    #return all recipes sorted by date of publication.
+    @recipes = Recipe.order(created_at: :desc).all.paginate(:page => params[:page], :per_page => 10)
```
Comment (moelheni) [inline app/controllers/recipe_controller.rb]: use a scope

Comment (achr3f) [inline app/controllers/recipe_controller.rb]: No need to `.all`  the `order()` will return a collection 
and plz remove  ruby 1.8 syntax

``` ruby
@recipes = Recipe.order(created_at: :desc).paginate( page: params[:page], per_page: 10 )
```

File: db/seeds.rb [hunk 2]
```diff
@@ -1,5 +1,22 @@
 
-#  Create an user
+[User, Ingredient, Recipe].each(&:delete_all)
```
Comment (achr3f) [inline db/seeds.rb]: space `[  User ..`

**Comment by SamiDghim** [comment]
@melas123 @achr3f @DoniaMroua @medBelaid plz check this code

File: app/controllers/recipe_controller.rb [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
   respond_to :json
+
   #Show all recipes
   def index
-    @recipes = Recipe.all
-    respond_with( @recipes )
+    #return all recipes sorted by date of publication.
+    @recipes = Recipe.most_recent.paginate( page: params[ :page ], per_page: 10 )
+    #byebug
+    respond_with( Paginator.pagination_attributes( @recipes ).merge!( :recipes => @recipes ), status: 200 )
```
Comment (melas123) [inline app/controllers/recipe_controller.rb]: plz use ruby 1.9 syntax 
respond_with( Paginator.pagination_attributes( @recipes ).merge!( recipes:  @recipes ), status: 200 )

Comment (SamiDghim) [inline app/controllers/recipe_controller.rb]: Done

**Pull request merged**